File: frontend/src/types/vehicle.ts
export interface VehicleData {
    speed: number;                 // Voertuigsnelheid (km/h)
    rpm: number;                   // Motortoerental (rpm)
    engineLoad: number;            // Motorbelasting (%)
    coolantTemperature: number;    // Koelvloeistoftemperatuur (°C)
    intakePressure: number;        // Inlaatdruk (kPa)
    intakeTemperature: number;     // Inlaatluchttemperatuur (°C)
    maf: number;                   // Luchtmassastroom (g/s)
    controlModuleVoltage: number;  // Spanning van de regelmodule (V)
    ambientTemperature: number;    // Buitentemperatuur (°C)

    // Optionele gegevens
    throttlePosition?: number;
    relativeThrottlePosition?: number;
    acceleratorPosition?: number;
    fuelLevel?: number;
    fuelRate?: number;
    fuelRailPressure?: number;
    barometricPressure?: number;
    distanceSinceDtcClear?: number;

    // Metadata voor performantiemeting
    sentAt?: number;
}

File: backend/models/vehicle.py
from pydantic import BaseModel
from typing import Optional


class VehicleData(BaseModel):
    # Basis voertuiggegevens
    speed: float
    rpm: float
    engineLoad: float
    coolantTemperature: float
    intakePressure: float
    intakeTemperature: float
    maf: float
    controlModuleVoltage: float
    ambientTemperature: float
    engineRunTime: float

    # Optionele gegevens (niet elke auto ondersteunt deze)
    throttlePosition: Optional[float] = None
    relativeThrottlePosition: Optional[float] = None
    acceleratorPosition: Optional[float] = None

    fuelLevel: Optional[float] = None
    fuelRate: Optional[float] = None
    fuelRailPressure: Optional[float] = None

    barometricPressure: Optional[float] = None

    distanceSinceDtcClear: Optional[float] = None

    # Metadata
    timestamp: Optional[float] = None
